packages/nuxt/src/Host.vue

```diff
@@ -91,33 +76,39 @@ const listen: ListenFunc = (listener) => {
   };
 };
 
-onBeforeMount(() => {
-  watchEffect(() => {
-    if (!cachedMount) {
-      loadScript(url)
-        .then(() => loadModule(name))
-        .then(({ default: createComposableApp }) => {
-          let remoteMount;
-          if (typeof createComposableApp === "function") {
-            const { mount } = createComposableApp({
-              isSelfHosted: false,
-            });
-            remoteMount = mount;
-          } else if (
-            typeof createComposableApp === "object" &&
-            createComposableApp.mount
-          ) {
-            remoteMount = createComposableApp.mount;
-          } else {
-            throw new Error("Remote module didn't return a function");
-          }
-          mount.value = remoteMount;
-          mountCache.set(mountKey, remoteMount);
-        })
-        .catch((err) => {
-          error.value = err;
+const HostWrapper = (mount: MountFunc, url?: string) => {
+  //@ts-ignore
+  return h(Mount, {
+    mount,
+    runtime,
+    basename,
+    listen,
+    setError: (e: any) => error.value = e,
+  });
+
+  
+};
+
+onMounted(() => {
+  watchEffect(async () => {
+    if (!cachedHostApp) {
+      try {
+        const { default: a } = await loadApp<Component>({
+          app: props.app,
+          remote: props.remote,
+          version,
+          context: {
+            origin: injectedOrigin,
+          },
+          HostWrapper,
         });
-    }
+        console.log(a);
+        mountCache.set(mountKey, a);
```

Comment:
I see we create the `a` variable because we can't use `default`. I was thinking of a more semantic name than `a` but I couldn't find anything, then I thought, would read it better if we don't create the `const`? e.g. 
```
 mountCache.set(
          mountKey,
          (
            await loadApp<Component>({
              app: props.app,
              remote: props.remote,
              version,
              context: {
                origin: injectedOrigin,
              },
              HostWrapper,
            })
          ).default
)
```